File: .claude/skills/sast-scan/rules/semgrep/go/tests/test_sql_injection.go
package main

import (
	"database/sql"
	"fmt"
	"os/exec"
)

func vulnerable(db *sql.DB, id string) {
	// Positive: fmt.Sprintf in Query
	// ruleid: go.security.sql-injection-string-concat
	db.Query(fmt.Sprintf("SELECT * FROM users WHERE id = %s", id))

	// ruleid: go.security.sql-injection-string-concat
	db.Query(fmt.Sprintf("SELECT * FROM products WHERE name = '%s'", name))

	// Positive: string concatenation in Exec
	// ruleid: go.security.sql-injection-string-concat
	db.Exec(fmt.Sprintf("DELETE FROM users WHERE id = %s", id))

	// ruleid: go.security.sql-injection-string-concat
	db.QueryRow(fmt.Sprintf("SELECT name FROM users WHERE id = %s", id))
}

func safe(db *sql.DB, id string) {
	// Negative: parameterized query with placeholder
	// ok: go.security.sql-injection-string-concat
	db.Query("SELECT * FROM users WHERE id = $1", id)

	// ok: go.security.sql-injection-string-concat
	db.Query("SELECT * FROM products WHERE name = ? AND active = ?", name, true)

	// ok: go.security.sql-injection-string-concat
	db.Exec("DELETE FROM users WHERE id = $1", id)

	// ok: go.security.sql-injection-string-concat
	db.QueryRow("SELECT name FROM users WHERE id = ?", id)
}

File: .claude/skills/sast-scan/rules/semgrep/java/tests/TestSqlInjection.java
import java.sql.*;

public class TestSqlInjection {
    public void vulnerable(Connection conn, String userId) throws SQLException {
        Statement stmt = conn.createStatement();

        // Positive: string concatenation in executeQuery
        // ruleid: java.security.sql-injection-string-concat
        stmt.executeQuery("SELECT * FROM users WHERE id = " + userId);

        // ruleid: java.security.sql-injection-string-concat
        stmt.executeQuery("SELECT * FROM users WHERE name = '" + userName + "'");

        // ruleid: java.security.sql-injection-string-concat
        stmt.execute("DELETE FROM users WHERE id = " + userId);

        // ruleid: java.security.sql-injection-string-concat
        stmt.executeUpdate("UPDATE users SET name = '" + userName + "' WHERE id = " + userId);

        // Positive: inline createStatement with concatenation
        // ruleid: java.security.sql-injection-string-concat
        conn.createStatement().executeQuery("SELECT * FROM users WHERE id = " + userId);
    }

    public void safe(Connection conn, String userId) throws SQLException {
        // Negative: PreparedStatement with parameterized query
        // ok: java.security.sql-injection-string-concat
        PreparedStatement pstmt = conn.prepareStatement("SELECT * FROM users WHERE id = ?");
        pstmt.setString(1, userId);

        // ok: java.security.sql-injection-string-concat
        PreparedStatement pstmt2 = conn.prepareStatement("SELECT * FROM users WHERE name = ? AND active = ?");
        pstmt2.setString(1, userName);
        pstmt2.setBoolean(2, true);
    }
}
